addons/gamification/models/badge.py

```diff
@@ -138,14 +138,24 @@ def _get_owners_info(self):
             the total number of time this badge was granted
             the total number of users this badge was granted to
         """
+        Users = self.env["res.users"]
+        query = Users._where_calc([])
+        Users._apply_ir_rules(query)
+        query.add_join(("res_users", "gamification_badge_user", "id", "user_id", "badges"))
+
+        tables, where_clauses, where_params = query.get_sql()
+
         self.env.cr.execute("""
-            SELECT badge_id, count(user_id) as stat_count,
-                count(distinct(user_id)) as stat_count_distinct,
-                array_agg(distinct(user_id)) as unique_owner_ids
-            FROM gamification_badge_user
-            WHERE badge_id in %s
-            GROUP BY badge_id
-            """, [tuple(self.ids)])
+              SELECT res_users__badges.badge_id, count(res_users.id) as stat_count,
+                     count(distinct(res_users.id)) as stat_count_distinct,
+                     array_agg(distinct(res_users.id)) as unique_owner_ids
+                FROM {tables}
+               WHERE {where_clauses}
+                 AND res_users__badges.badge_id in %s
+            GROUP BY res_users__badges.badge_id
+            """.format(tables=tables, where_clauses=where_clauses),
```

Comment:
you can use f-string instead of `format`, but OK.

Reply:
yep, true! => done :+1: